### DISCUSSION TITLE
how to split get_svg_images() or get each individual image or curve

### DISCUSSION THREAD

**Discussion by zdw1011781461** [pymupdf/PyMuPDF#1097] [Q&A]
I have a PDF file with images and curves in it. I use get_images() to get the image data but not the clipPath in SVG , How can I get the clipPath separately?
[test.pdf](https://github.com/pymupdf/PyMuPDF/files/6674238/test.pdf)

**Comment by JorjMcKie** [comment]
For this I would have to extend `page.get_drawings()`. Currently, I am simply ignoring any clippings.
So you would have to write your own analyzer of the `page.get_svg_image()` XML.

**Reply by zdw1011781461** [reply]
What you mean is get_svg_images parses each one by regular or other methods?

**Reply by JorjMcKie** [reply]
Yes, `get_svg_image()` is XML. You could use the [lxml](https://pypi.org/project/lxml/) package.
Another approach is using the output of the MuPDF CLI tool `mutool trace test.pdf`. This also is XML. Actually both, `get_svg_image()` and `get_drawings()` are based on this.
Although SVG extraction is provided by MuPDF, whereas `get_drawings()` is my own making.

**Reply by zdw1011781461** [reply]
Thanks for the advice
